controller/src/main/java/io/pravega/controller/store/stream/AbstractStreamMetadataStore.java

```diff
@@ -187,7 +189,16 @@ public OperationContext createStreamContext(String scopeName, String streamName,
                         Futures.completeOn(checkScopeExists(scope, context, executor)
                                 .thenCompose(exists -> {
                                     if (exists) {
-                                        // Create stream may fail if scope is deleted as we attempt to create the stream under scope. 
+                                        checkScopeInDeletingTable(scope, context, executor)
+                                                .thenCompose(exist -> {
+                                                    if (exist) {
+                                                        throw new ScopeDeletionInProgressException("Scope already in deleting state: " + scope);
```

Comment:
@shshashwat I thought we discussed returning FailedFuture with embedded UnsupportedOperationException instead.
I don;t see the need to create a new exception class here

Reply:
This new exception was created mainly to give a correct status on client. The new exception is extending to UnsupportedOperationException and during #6446 we will handle to check if the type of exception thrown from Controller is `ScopeDeletionInProgressException` then we would update the client accordingly. Right now it will give a controller failure exception which is not correct.

Reply:
Attempting to create a Stream inside a scope that is in deleting state is a normal operation and not an exceptional situation. Exceptions are expensive and so should **not** be used in areas where we can do with returning a failure status indicating why the operation could not be completed. Here instead of throwing exception, we should return a new status CreateStreamResponse.Status=SCOPE_SEALED.

Reply:
`CreateStreamResponse` doesn not have a state `SCOPE_SEALED`. If required may be we can make required changes as part of enhancement.